remote-config/src/fetch/fetcher.rs

```diff
@@ -0,0 +1,316 @@
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::{Arc, Mutex, MutexGuard};
+use base64::Engine;
+use hyper::http::uri::{PathAndQuery, Scheme};
+use hyper::{Body, Client, StatusCode};
+use sha2::{Digest, Sha256, Sha512};
+use tracing::{debug, trace, warn};
+use datadog_trace_protobuf::remoteconfig::{ClientGetConfigsRequest, ClientGetConfigsResponse, ClientState, ClientTracer, ConfigState, TargetFileHash, TargetFileMeta};
+use ddcommon::{connector, Endpoint};
+use crate::{RemoteConfigPath, Target};
+use crate::targets::TargetsList;
+
+const PROD_INTAKE_SUBDOMAIN: &str = "config";
+
+/// Manages files.
+/// Presents store() and update() operations.
+/// It is recommended to minimize the overhead of these operations as they
```

Comment:
I think this comment is incomplete

Reply:
... will be invoked while a lock across all ConfigFetchers referencing the same ConfigFetcherState is held.

Will complete :-)